# EmKa Academy

## UYARI ! : Lütfen çıktının olduğu kod hücresini çalıştırmayın, aksi takdirde almanız gereken çıktı görüntüsünü kaybedebilirsiniz.

- Evrensel kısaltmaları kullanarak numpy ve pandas kütüphanelerini import ediniz.

1) [GitHubda](https://github.com/EmKa-Academy/Udemy-Python_Numpy_Pandas) bulunan "Advertising.csv" reklam datasını herhangi bir platformda DF olarak okuyun ve "df" adında bir değişkene atayın. Ardından df değişkeninin ilk 3 satırına, son 3 satırına , rastgele 3 satırına bakın.

In [ ]:
# Buraya kodunuzu yazın

In [2]:
# Buraya kod yazmayın !

TV  radio  newspaper  sales
0  230.1   37.8       69.2   22.1
1   44.5   39.3       45.1   10.4
2   17.2   45.9       69.3    9.3

In [ ]:
# Buraya kodunuzu yazın

In [3]:
# Buraya kod yazmayın !

TV  radio  newspaper  sales
197  177.0    9.3        6.4   12.8
198  283.6   42.0       66.2   25.5
199  232.1    8.6        8.7   13.4

In [ ]:
# Buraya kodunuzu yazın(Rastgele 3 satır çekileceği için sizde aşağıdaki 3 satırın aynısı gelmeyecektir. Mucizeye yakın bir ihtimal gerçekleşmeyecekse :D )

In [4]:
# Buraya kod yazmayın !

TV  radio  newspaper  sales
180  156.6    2.6        8.3   10.5
69   216.8   43.9       27.2   22.3
12    23.8   35.1       65.9    9.2

2) Elde ettiğiniz "df" in infosuna ve describe'ına bakın ve yorumlamaya çalışın.

In [5]:
# Buraya kod yazmayın !

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 4 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   TV         200 non-null    float64
 1   radio      200 non-null    float64
 2   newspaper  200 non-null    float64
 3   sales      200 non-null    float64
dtypes: float64(4)
memory usage: 6.4 KB


In [6]:
# Buraya kod yazmayın !

TV       radio   newspaper       sales
count  200.000000  200.000000  200.000000  200.000000
mean   147.042500   23.264000   30.554000   14.022500
std     85.854236   14.846809   21.778621    5.217457
min      0.700000    0.000000    0.300000    1.600000
25%     74.375000    9.975000   12.750000   10.375000
50%    149.750000   22.900000   25.750000   12.900000
75%    218.825000   36.525000   45.100000   17.400000
max    296.400000   49.600000  114.000000   27.000000

3) df içinden sadece "TV" sütununu seri olarak çekin

In [7]:
# Buraya kod yazmayın !

0      230.1
1       44.5
2       17.2
3      151.5
4      180.8
       ...  
195     38.2
196     94.2
197    177.0
198    283.6
199    232.1
Name: TV, Length: 200, dtype: float64

4) df içinden "TV" ve "sales" sütununu çekin ve ekrana yazdırın

In [8]:
# Buraya kod yazmayın !

TV  sales
0    230.1   22.1
1     44.5   10.4
2     17.2    9.3
3    151.5   18.5
4    180.8   12.9
..     ...    ...
195   38.2    7.6
196   94.2    9.7
197  177.0   12.8
198  283.6   25.5
199  232.1   13.4

[200 rows x 2 columns]

5) df içindeki "TV" , "radio" , "newspaper" sütunlarındaki değerleri toplayarak "total_adv" adında yeni bir sütun oluşturun ve df'i ekrana yazdırın.

TV  radio  newspaper  sales  total_adv
0  230.1   37.8       69.2   22.1      337.1
1   44.5   39.3       45.1   10.4      128.9
2   17.2   45.9       69.3    9.3      132.4
3  151.5   41.3       58.5   18.5      251.3
4  180.8   10.8       58.4   12.9      250.0

6) df_adv içindeki "sales" sütununun drop edilmiş halini ekrana yazdırın. (Kalıcı hale getirmeyin. Sadece ekrana bastırın)

In [11]:
# Buraya kod yazmayın !

TV  radio  newspaper  total_adv
0    230.1   37.8       69.2      337.1
1     44.5   39.3       45.1      128.9
2     17.2   45.9       69.3      132.4
3    151.5   41.3       58.5      251.3
4    180.8   10.8       58.4      250.0
..     ...    ...        ...        ...
195   38.2    3.7       13.8       55.7
196   94.2    4.9        8.1      107.2
197  177.0    9.3        6.4      192.7
198  283.6   42.0       66.2      391.8
199  232.1    8.6        8.7      249.4

[200 rows x 4 columns]

7) df içindeki son 20 index'i(satırı) çekin ve ekrana yazdırın.

In [12]:
# Buraya kod yazmayın !

TV  radio  newspaper  sales  total_adv
180  156.6    2.6        8.3   10.5      167.5
181  218.5    5.4       27.4   12.2      251.3
182   56.2    5.7       29.7    8.7       91.6
183  287.6   43.0       71.8   26.2      402.4
184  253.8   21.3       30.0   17.6      305.1
185  205.0   45.1       19.6   22.6      269.7
186  139.5    2.1       26.6   10.3      168.2
187  191.1   28.7       18.2   17.3      238.0
188  286.0   13.9        3.7   15.9      303.6
189   18.7   12.1       23.4    6.7       54.2
190   39.5   41.1        5.8   10.8       86.4
191   75.5   10.8        6.0    9.9       92.3
192   17.2    4.1       31.6    5.9       52.9
193  166.8   42.0        3.6   19.6      212.4
194  149.7   35.6        6.0   17.3      191.3
195   38.2    3.7       13.8    7.6       55.7
196   94.2    4.9        8.1    9.7      107.2
197  177.0    9.3        6.4   12.8      192.7
198  283.6   42.0       66.2   25.5      391.8
199  232.1    8.6        8.7   13.4      249.4

8) df içinde satışları 25'den fazla olanları getirin

In [13]:
# Buraya kod yazmayın !

TV  radio  newspaper  sales  total_adv
36   266.9   43.8        5.0   25.4      315.7
98   289.7   42.3       51.2   25.4      383.2
147  243.2   49.0       44.3   25.4      336.5
175  276.9   48.9       41.8   27.0      367.6
183  287.6   43.0       71.8   26.2      402.4
198  283.6   42.0       66.2   25.5      391.8

9) df içinde "TV" reklamlarına 200'den fazla VE "newspaper" reklamlarına 50'dan fazla harcama yapanları getirin.

In [14]:
# Buraya kod yazmayın !

TV  radio  newspaper  sales  total_adv
0    230.1   37.8       69.2   22.1      337.1
17   281.4   39.6       55.8   24.4      376.8
20   218.4   27.7       53.4   18.0      299.5
61   261.3   42.7       54.7   24.2      358.7
92   217.7   33.5       59.0   19.4      310.2
93   250.9   36.5       72.3   22.2      359.7
98   289.7   42.3       51.2   25.4      383.2
101  296.4   36.3      100.9   23.8      433.6
110  225.8    8.2       56.5   13.4      290.5
124  229.5   32.3       74.2   19.7      336.0
137  273.7   28.9       59.7   20.8      362.3
165  234.5    3.4       84.8   11.9      322.7
168  215.4   23.6       57.6   17.1      296.6
183  287.6   43.0       71.8   26.2      402.4
198  283.6   42.0       66.2   25.5      391.8

10) df içinde "radio" reklamlarına 10'dan az ve "TV" reklamlarına 250'den çok harcama yapanları getirin

In [15]:
# Buraya kod yazmayın !

TV  radio  newspaper  sales  total_adv
25   262.9    3.5       19.5   12.0      285.9
35   290.7    4.1        8.5   12.8      303.3
131  265.2    2.9       43.0   12.7      311.1
178  276.7    2.3       23.7   11.8      302.7

11) df içinde 'TV' reklam harcamaları 10'dan az ya da eşit olan VEYA 'newspaper' harcamaları 100'den fazla olanları getirin.

In [16]:
# Buraya kod yazmayın !

TV  radio  newspaper  sales  total_adv
5      8.7   48.9       75.0    7.2      132.6
8      8.6    2.1        1.0    4.8       11.7
16    67.8   36.6      114.0   12.5      218.4
56     7.3   28.1       41.4    5.5       76.8
78     5.4   29.9        9.4    5.3       44.7
101  296.4   36.3      100.9   23.8      433.6
126    7.8   38.9       50.6    6.6       97.3
130    0.7   39.6        8.7    1.6       49.0
132    8.4   27.2        2.1    5.7       37.7
155    4.1   11.6        5.7    3.2       21.4

12) df içinde 'sales' miktarı 20'den fazla ve 'total_adv' miktarı 250'den az olanları getirin.

In [17]:
# Buraya kod yazmayın !

TV  radio  newspaper  sales  total_adv
139  184.9   43.9        1.7   20.7      230.5

13) df içinde 'sales' ları 23 ile 25(23 ve 25 dahil) arasında olanları getirin

In [18]:
# Buraya kod yazmayın !

TV  radio  newspaper  sales  total_adv
17   281.4   39.6       55.8   24.4      376.8
47   239.9   41.5       18.5   23.2      299.9
55   198.9   49.4       60.0   23.7      308.3
58   210.8   49.6       37.7   23.8      298.1
61   261.3   42.7       54.7   24.2      358.7
101  296.4   36.3      100.9   23.8      433.6
128  220.3   49.0        3.2   24.7      272.5

14) df içinde 'salesları tam olarak 18.3 veya 14.0 veya 25.5 veya 21.7 veya 13.6 veya 6.9 veya 23.8 olanları getirin.(Not:Bunu yapmanın 2 yolu vardı, basi olanıyla yapmaya çalışın)

In [19]:
# Buraya kod yazmayın !

TV  radio  newspaper  sales  total_adv
58   210.8   49.6       37.7   23.8      298.1
63   102.7   29.6        8.4   14.0      140.7
70   199.1   30.6       38.7   18.3      268.4
76    27.5    1.6       20.7    6.9       49.8
83    68.4   44.5       35.6   13.6      148.5
84   213.5   43.0       33.8   21.7      290.3
101  296.4   36.3      100.9   23.8      433.6
198  283.6   42.0       66.2   25.5      391.8

15) df içinde 'sales' ları 20'den büyük olanların oranı kaç ?

In [20]:
# Buraya kod yazmayın !

0.155

In [ ]:
# Done.